c10/core/DispatchKey.h

```diff
@@ -19,47 +27,85 @@ enum class DispatchKey : uint8_t {
   // and Undefined == nullopt.  We didn't actually represent
   // it this way because optional<RealDispatchKey> would take two
   // words, when DispatchKey fits in eight bits.
+
   Undefined = 0,
 
-  // This pool of IDs is not really ordered, but it is merged into
-  // the hierarchy for convenience and performance
-  CPUTensorId, // PyTorch/Caffe2 supported
-  CUDATensorId, // PyTorch/Caffe2 supported
-  MKLDNNTensorId, // Caffe2 only
-  OpenGLTensorId, // Caffe2 only
-  OpenCLTensorId, // Caffe2 only
-  IDEEPTensorId, // Caffe2 only
-  HIPTensorId, // PyTorch/Caffe2 supported
-  SparseHIPTensorId, // PyTorch only
-  MSNPUTensorId, // PyTorch only
-  XLATensorId, // PyTorch only
-  MkldnnCPUTensorId,
-  QuantizedCPUTensorId, // PyTorch only
-  ComplexCPUTensorId, // PyTorch only
-  ComplexCUDATensorId, // PyTorch only
-
-  // See Note [Private use TensorId]
-  PrivateUse1_TensorId,
-  PrivateUse2_TensorId,
-  PrivateUse3_TensorId,
 
-  // Sparse has multi-dispatch with dense; handle it first
-  SparseCPUTensorId, // PyTorch only
-  SparseCUDATensorId, // PyTorch only
 
-  // Custom pseudorandom number generator dispatch key.
-  // To enable PyTorch users to have custom PRNGs, we added support for
-  // dispatching on at::Generator* parameters.
-  // This key must be used in two places:
-  //  1) as a second parameter of at::Generator constructor call in
+  // ~~~~~~~~~~~~~~~~~~~~~~~~~~ BACKENDS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
+  // A "backend" is colloquially used to refer to handlers for dispatch
+  // which actually implement the numerics of an operation in question.
+  // Alternately, it is placed in the "back" of the dispatch hierarchy,
```

Comment:
This sentence seems pretty questionable -- it wouldn't surprise me at all if we have some things running after the "backend" in the future (for tracking purposes on the output) and I don't think that's what people mean at all when they say "backend" (they just mean, it's the thing that provides the implementation for the frontend).